redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/WidgetSystem.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.treehouse
+
+import app.cash.redwood.protocol.widget.DiffConsumingNode
+import app.cash.redwood.protocol.widget.ProtocolMismatchHandler
+import kotlinx.serialization.json.Json
+
+public fun interface WidgetSystem {
+  /** Returns a widget factory for encoding and decoding changes of a [Content]. */
+  public fun widgetFactory(
+    app: TreehouseApp<*>,
+    json: Json,
```

Comment:
It wasn’t good for TreehouseView to depend on JSON and protocol stuff. I promoted this to a top-level class to later support moving TreehouseView into a different module; perhaps with a different name.